src/coreclr/vm/codeman.cpp

```diff
@@ -3158,7 +3158,11 @@ JumpStubBlockHeader *  EEJitManager::allocJumpStubBlock(MethodDesc* pMD, DWORD n
         ExecutableWriterHolder<CodeHeader> codeHdrWriterHolder(pCodeHdr, sizeof(CodeHeader));
         codeHdrWriterHolder.GetRW()->SetStubCodeBlockKind(STUB_CODE_BLOCK_JUMPSTUB);
 
-        NibbleMapSetUnlocked(pCodeHeap, mem, TRUE);
+        if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_UseOptimizedNibbleMap)){
+            NibbleMapSetUnlocked2(pCodeHeap, mem, sizeof(CodeHeader) + blockSize + (CODE_SIZE_ALIGN - 1));
+        } else {
+            NibbleMapSetUnlocked(pCodeHeap, mem, TRUE);
```

Comment:
Maybe make a pair of helper classes/namespaces and put the bit fiddling implementations in there?